### PR TITLE
Doc review of Global Configuration doc

### LINKED ISSUE #97: DOC: Review Global Configuration in Spartacus
Conduct an in-depth review of https://github.com/SAP/cloud-commerce-spartacus-storefront/wiki/Global-Configuration-in-Spartacus-(DRAFT)

This input comes from GH-2027, and is linked to GH-1347.

### PR DESCRIPTION
**Pull request by gladius-mtl**
Closes #97

### CODE REVIEW

File: _pages/dev/global-configuration-in-spartacus.md [hunk 1]
```diff
@@ -17,30 +15,29 @@ Any configuration provided can be overridden in the main app module.
...
+If you import the `B2cStorefrontModule` in your app, the preferred and easiest method to provide a configuration is to use `B2cStorefrontModule.withConfig(config?: StorefrontModuleConfig)`. The `StorefrontModuleConfig` interface provides type safety and context-aware code completion that speeds up defining the configuration and helps to avoid any typos.
 
 ### ConfigModule.withConfig
 
-Importing `ConfigModule.withConfig(config: any)` is useful, when we want to use the configuration in our module and, at the same time, contribute to it.
-`config` parameter is not typed, can be an object of any shape, so it suggested to either use typed const or use type casting `ConfigModule.withConfig(<ConfigType>{})` to utilize type safety.
+Importing `ConfigModule.withConfig(config: any)` is useful when you want to use the configuration in your module and contribute to it at the same time. The `config` parameter is not typed and can be an object of any shape, so it is recommended that you use either a typed `const`, or that you use `ConfigModule.withConfig(<ConfigType>{})` type casting to be able to use type safety.
 
 ### provideConfig
```
Comment (dunqan) [inline _pages/dev/global-configuration-in-spartacus.md:30]: ProvideConfig should be used as preffered one, `ConfigModule.withConfig` as the legacy one, and `(B2c)StorefrontModule.withConfig` are dropped already.

File: _pages/dev/global-configuration-in-spartacus.md [hunk 2]
```diff
@@ -2,11 +2,9 @@
 title: Global Configuration in Spartacus
 ---
 
-Spartacus uses a configuration mechanism that provides global app configuration at app bootstrap and is not changed while the app is running.
+Spartacus uses a mechanism that provides global configuration during app initialization (that is, when the application is bootstrapped). This configuration does not change while the application is running. Each storefront module that uses this configuration usually provides typing with some defaults for its part of the configuration.
 
-Each storefront module that is using configuration usually provides typing for its part of the configuration with some defaults.
-
-Any configuration provided can be overridden in the main app module.
+If necessary, you can always override the global configuration in the main app module.
```
[the remark this answers is not in the document]
Comment (gladius-mtl) [reply _pages/dev/global-configuration-in-spartacus.md:7]: Ah, I see! Based on your comment, I am changing the line into a note:

>Note: The configuration in the main app module takes precedence over other configurations, and can be used to override any configuration that has been provided elsewhere.

File: _pages/dev/global-configuration-in-spartacus.md [hunk 3]
```diff
@@ -51,116 +48,103 @@ i18n: {
...
-## Overriding values
+Each module typically provides a default configuration that is required for basic functionality. A default configuration is any configuration that is provided inside the related module. It usually contains reasonable defaults or configurations that are required for a module to operate.
 
-Configuration mechanism used in spartacus is built upon and subject to the rules of standard Angular Dependency Injection mechanism,
-Each configuration chunk (default or override) is provided separately, with multi-provider feature and all chunks are merged
-in a factory used to inject configuration.
+You do not need to provide defaults for all required configurations. For example, it may be difficult to choose reasonable defaults for some options. In this case, it is recommended to use config validators to validate the configuration and warn (in development mode) if a required config is missing.
 
-### How merging process works?
+**Note:** Default configurations are provided in the same way as any other configuration.
```
[the remark this answers is not in the document]
Comment (gladius-mtl) [reply _pages/dev/global-configuration-in-spartacus.md:67]: Note changed to reflect this info 🙂

File: _pages/dev/global-configuration-in-spartacus.md (hunk 1 above)
Comment (gladius-mtl) [reply _pages/dev/global-configuration-in-spartacus.md:30]: I have changed the order of the sections to reflect your comment, I have pretty much removed the contents of the `StorefrontModule.withConfig` section (except to say that it is no longer supported), and I have added a line at the beginning of the other sections to say which one is the preferred method and which one is the legacy method.

Beyond that, I assume we should modify what the text currently says about `provideConfig` and `ConfigModule.withConfig`. Do you have any suggestions? Or is there somone I can follow up with to ask for a "rewrite" of these sections? 🙂

File: _pages/dev/global-configuration-in-spartacus.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (dunqan) [reply _pages/dev/global-configuration-in-spartacus.md]: 👍

File: _pages/dev/global-configuration-in-spartacus.md [hunk 4]
```diff
@@ -17,30 +15,33 @@ Any configuration provided can be overridden in the main app module.
...
+If you want to contribute to the global configuration without needing to import the `ConfigModule`, or you want to implement a module with providers and provide the configuration conditionally, then using `provideConfig` in a providers array is the best option.
 
 ### ConfigModule.withConfig
 
-Importing `ConfigModule.withConfig(config: any)` is useful, when we want to use the configuration in our module and, at the same time, contribute to it.
-`config` parameter is not typed, can be an object of any shape, so it suggested to either use typed const or use type casting `ConfigModule.withConfig(<ConfigType>{})` to utilize type safety.
+Using `ConfigModule.withConfig` is the legacy method of providing global configuration in Spartacus.
 
-### provideConfig
+Importing `ConfigModule.withConfig(config: any)` is useful when you want to use the configuration in your module and contribute to it at the same time. The `config` parameter is not typed and can be an object of any shape, so it is recommended that you use either a typed `const`, or that you use `ConfigModule.withConfig(<ConfigType>{})` type casting to be able to use type safety.
```
Comment (dunqan) [inline _pages/dev/global-configuration-in-spartacus.md:32]:
```suggestion
Importing `ConfigModule.withConfig(config: Config)` is useful when you want to use the configuration in your module and contribute to it at the same time. 
```

File: _pages/dev/global-configuration-in-spartacus.md (hunk 1 above)
Comment (dunqan) [reply _pages/dev/global-configuration-in-spartacus.md]: 👍

File: _pages/dev/global-configuration-in-spartacus.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (dunqan) [reply _pages/dev/global-configuration-in-spartacus.md]: 👍

File: _pages/dev/global-configuration-in-spartacus.md [hunk 5]
```diff
@@ -17,30 +15,33 @@ Any configuration provided can be overridden in the main app module.
...
-### StorefrontLib.withConfig
+The following sections describe the different ways you can provide global configuration in Spartacus.
+
+### provideConfig
 
-If you import B2cStorefrontModule in your app, preferred and easiest method to provide configuration is to use `B2cStorefrontModule.withConfig(config?: StorefrontModuleConfig)`.
-`StorefrontModuleConfig` interface provides type safety and context-aware code completion that speed ups defining configuration and helps to avoid any typos.
+Using `provideConfig` is the preferred way for providing global configuration in Spartacus.
+
+If you want to contribute to the global configuration without needing to import the `ConfigModule`, or you want to implement a module with providers and provide the configuration conditionally, then using `provideConfig` in a providers array is the best option.
```
Comment (dunqan) [inline _pages/dev/global-configuration-in-spartacus.md:26]:
```suggestion
If you want to contribute to the global configuration using `provideConfig`, just add it to a providers array in an Angular module.
```

**Pull request merged**